lib/configure.js

```diff
@@ -115,7 +115,7 @@ function configure (gyp, argv, callback) {
 
     log.verbose('build/' + configFilename, 'creating config file')
 
-    var config = process.config || {}
+    var config = JSON.parse(JSON.stringify(process.config)) || {}
       , defaults = config.target_defaults
```

Comment:
`JSON.parse(JSON.stringify(null))` becomes `"null"`, which is not falsey, so the code no longer falls back to `{}`. Below should retain the original intent, while also covering your change.

Also added in a comment to explain the need to stringify/pars.

```suggestion
    // In Node v19 process.config is now frozen, so we need to clone it here:
    var config = JSON.parse(JSON.stringify(process.config || {}))

    var defaults = config.target_defaults
```

Reply:
Thanks for repsonse, done.